02_commands.html

```diff
@@ -1,206 +1,524 @@
+
+    <p>After such a <a>connection</a> has been established, a remote end MUST run the following steps:</p>
+
+    <ol>
+      <!-- Should mention what happens if it can't be decoded as HTTP -->
+      <!-- Note: the way that fetch is used here isn't quite right because it doesn't
+           specify how to construct a request from network data, or serialize a response. -->
+
+      <li><p><a>Read bytes</a> from the connection until a complete HTTP request can be constructed from the data. Let <var>request</var> be a <a href="https://fetch.spec.whatwg.org/#concept-request" class="externalRef">request</a> object constructed from the received data, according to the requirements of [[!RFC7230]].</p></li>
+
+      <li><p>Let <var>request match</var> be the result of <a>matching a request</a> with <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-method" class="externalRef">method</a> and <a href="https://fetch.spec.whatwg.org/#concept-request-url" class="externalRef">url</a> as arguments.</p></li>
+
+      <li><p>If <var>request match</var> is of type <a>error</a>, <a>return an error</a> with <var>request match</var>'s <a>error code</a> and jump to step 1.</p>
+
+        <p>Otherwise, let <var>command</var>, <var>session id</var> and <var>element id</var> be <var>request match</var>'s data.</p></li>
+
+      <li><p>If <var>command</var> is not <a>new session</a>:</p>
+        <ol>
+          <li><p>If <var>session id</var> is not equal to the <a>id</a> of any session in the <a>list of active sessions</a>, <a>return an error</a> with <a>error code</a> <a>invalid session id</a>, then jump to step 1.</p>
+
+          <p>Otherwise, let the <a>current session</a> be the session with <a>id</a> <var>session id</var></p>
+          </li>
+        </ol>
+      </li>
+
+      <li><p>If <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-method" class="externalRef">method</a> is POST:</p>
+        <ol>
+
+          <li><p>Let <var>success</var> and <var>parameters</var> be the result of <a>parsing as JSON</a> with <var>request</var>'s <a href="https://fetch.spec.whatwg.org/#concept-request-body" class="externalRef">body</a> as the argument.</p></li>
+
+          <li><p>If <var>success</var> is false, or <var>success</var> is true and <var>parameters</var> is not an Object object, <a>return an error</a> with status <a>error code</a> <a>invalid argument</a> and jump back to step 1 in this overall algorithm.</p></li>
+        </ol>
+
+        <p>Otherwise, let <var>parameters</var> be null</p>
+      </li>
+
+      <li><p>Let <var>response data</var> be a <a>command response</a> object obtained by running running the steps defined by <var>command</var> with arguments <var>element id</var> and <var>parameters</var>.</p></li>
+
+      <li><p>If <var>response data</var> is an <a>error</a>, <a>return an error</a> with <a>error code</a> equal to <var>response data</var>'s <a>error code<a>.</p>
```

Comment:
I'm a little unclear on what it means to _be_ an `error` here, but it reads nice in English.  E.g. better than saying if `response data` has `error code` defined, then take that an map it to the appropriate `error code`.